src/DataProtection/DataProtection/src/KeyManagement/KeyManagementOptions.cs

```diff
@@ -94,6 +94,25 @@ internal static TimeSpan MaxServerClockSkew
         }
     }
 
+    /// <summary>
+    /// During each default key resolution, if a key decryption attempt fails,
+    /// it can be retried, as long as the total number of retries across all keys
+    /// does not exceed this value.
+    /// </summary>
+    /// <remarks>
+    /// Settable for testing.
+    /// </remarks>
+    internal int MaximumDefaultKeyResolverRetries { get; set; } = 10;
```

Comment:
This should probably drop to 1 or 0 if `Microsoft.AspNetCore.DataProtection.KeyManagement.DisableAsyncKeyRingUpdate` is set (from https://github.com/dotnet/aspnetcore/pull/54675)

Reply:
I'm wondering if we actually want a separate appcontext switch for this so people can tune the value.

Reply:
I think I'll replace the existing contextswitch with a numeric one and let 0 mean "disable both #54711 and #54675".

Reply:
Changed my mind again - I think two switches is preferable so that people can adjust them independently, as they see fit.  To whatever extent we provide guidance around appcontext switches, we can discourage using one without the other.